permission_handler_apple/ios/Classes/strategies/EventPermissionStrategy.m

```diff
@@ -37,49 +37,113 @@ - (void)requestPermission:(PermissionGroup)permission completionHandler:(Permiss
     
     EKEntityType entityType;
     
-    if (permission == PermissionGroupCalendar) {
+    if (permission == PermissionGroupCalendar || permission == PermissionGroupCalendarWriteOnly) {
         #if PERMISSION_EVENTS
         entityType = EKEntityTypeEvent;
         #else
         completionHandler(PermissionStatusDenied);
         return;
         #endif
+    } else if (permission == PermissionGroupCalendarFullAccess) {
+        #if PERMISSION_EVENTS_FULL_ACCESS
+        entityType = EKEntityTypeEvent;
+        #else
+        completionHandler(PermissionStatusDenied);
+        return;
+        #endif
     } else if (permission == PermissionGroupReminders) {
         #if PERMISSION_REMINDERS
         entityType = EKEntityTypeReminder;
         #else
         completionHandler(PermissionStatusDenied);
         return;
         #endif
-    } else {
+    }
+
+    else {
         completionHandler(PermissionStatusPermanentlyDenied);
         return;
     }
-    
+
     EKEventStore *eventStore = [[EKEventStore alloc] init];
-    [eventStore requestAccessToEntityType:entityType completion:^(BOOL granted, NSError *error) {
-        if (granted) {
-            completionHandler(PermissionStatusGranted);
+
+    if (@available(iOS 17.0, *)) {
+        if (entityType == EKEntityTypeEvent) {
+            #if PERMISSION_EVENTS_FULL_ACCESS
```

Comment:
Forgive me as my Objective-C knowledge could use some polishing, but would this work if the application wants the option to ask for either permission? Let's say `PERMISSION_EVENTS_FULL_ACCESS` and `PERMISSION_EVENTS` are both `1`, and the application requests `Permission.calendarWriteOnly`.